src/components/QuoteGenerator.vue

```diff
@@ -1,132 +1,204 @@
 <template>
   <form>
     <div class='form-value__container'>
-      <label for='num-of-doors'>Number of doors:</label>
-      <input v-model='doors' type='number' min='1' max='1000' id='num-of-doors' required/>
-      <small>Warning message</small>
+      <form-input
+        required
+        label='Number of doors'
+        placeholder='Number of doors'
+        :value='doors'
+        :min='1'
+        :max='1000'
+        @change='(...v) => updateDoorsValue(...v)'
+      >
+      </form-input>
     </div>
 
     <div class='form-value__container'>
-      <input id='bulk-panels' type='checkbox' v-model='isPanelsOnlyOrder'/>
-      <label for='bulk-panels'>Panels Only (Bulk)</label>
+      <control-input
+        label='Panels Only (Bulk)'
+        :checked='isPanelsOnlyOrder'
+        @toggle='isPanelsOnlyOrder = !isPanelsOnlyOrder'
+      >
+      </control-input>
     </div>
 
     <div class='form-value__container'>
-      <label for='door-width'>Door width:</label>
-      <select v-model='selectedDoorWidth' id='door-width'>
-        <option v-for='width in displayDoorWidths' :key='`w-key-${width}`'>{{ width }}</option>
-      </select>
+      <form-select
+        label='Door width'
+        :options='displayDoorWidths'
+        :selected='selectedDoorWidth'
+        @select='(...v) => { selectedDoorWidth = v[0] }'
+      >
+      </form-select>
     </div>
 
     <div class='form-value__container'>
-      <label for='door-height'>Door height:</label>
-      <select v-model='selectedDoorHeight' id='door-height'>
-        <option v-for='height in displayDoorHeights' :key='`h-key-${height}`'>{{ height }}</option>
-      </select>
+      <form-select
+        label='Door height'
+        :options='displayDoorHeights'
+        :selected='selectedDoorHeight'
+        @select='(...v) => { selectedDoorHeight = v[0] }'
+      >
+      </form-select>
     </div>
 
     <div class='form-value__container'>
-      <label for='stamp-pattern'>Stamp Pattern:</label>
-      <select v-model='selectedStampPattern' id='stamp-pattern'>
-        <option v-for='pattern in stampPatterns' :key='pattern.key' :value='pattern.key'>{{ pattern.displayName }}</option>
-      </select>
+      <form-select
+        label='Stamp Pattern'
+        :options='stampPatterns.map((s) => s.displayName)'
+        :values='stampPatterns.map((s) => s.key)'
```

Comment:
Values to be mapped to models/order.ts
```
<option v-for='pattern in stampPatterns' 
:key='pattern.key' 
:value='pattern.value'>
{{ pattern.displayName }}</option>
```

Reply:
Will wait for https://github.com/Sourav-Patteri/UpwardorPortalVue/pull/4 to be merged first, to use the updated model.